runtime/alloc.c

```diff
@@ -371,3 +371,27 @@ CAMLexport value caml_alloc_some(value v)
   Field(some, 0) = v;
   CAMLreturn(some);
 }
+
+
+/* For aligned atomic */
+
+CAML_STATIC_ASSERT((sizeof(value)) <= 8);
```

Comment:
Where does the "8" come from? Why is this important here? 

Generally, it would be best to avoid hard-coded constants in the code.

Reply:
Removed this assertion altogether. It doesn't seem needed, in hindsight.